### PR TITLE
add husky with pre-commit lint hook

### PR DESCRIPTION
**Pull request by alxnddr**
### Description

Add [husky](https://github.com/typicode/husky) with `pre-commit` hook that runs prettier on staged filed.

How to test:
Try to commit files with invalid formatting and ensure prettier has been executed

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,6 +26,7 @@
     "grid-styled": "4.1.0",
     "history": "3",
     "humanize-plus": "^1.8.1",
+    "husky": "^4.0.0",
```
Comment (kulyk) [inline package.json:29]: Just out of curiosity, why not using the latest version (6)?

**Review by kulyk — APPROVED** [review]
Thank you 🙌

File: package.json (hunk 1 above)
Comment (alxnddr) [reply package.json]: Oh, they slightly changed API, updated it to use 6 version. Thanks for pointing out.

File: package.json [hunk 2]
```diff
@@ -203,7 +204,8 @@
     "test-cypress-open": "./bin/build-for-test && yarn test-cypress-no-build --open",
     "test-cypress-open-no-backend": "E2E_HOST='http://localhost:3000' yarn test-cypress-no-build --open",
     "test-cypress-no-build": "yarn && CONFIG_FILE=frontend/test/__support__/e2e/cypress.json babel-node ./frontend/test/__runner__/run_cypress_tests.js",
-    "test-cypress-smoketest": "yarn test-cypress-no-build --folder frontend/test/metabase-smoketest"
+    "test-cypress-smoketest": "yarn test-cypress-no-build --folder frontend/test/metabase-smoketest",
+    "prepare": "husky install"
```
Comment (nemanjaglumac) [inline package.json:208]: Can this script name be a bit more specific, please?
"install-husky", or anything similar.

Comment (alxnddr) [reply package.json:208]: @nemanjaglumac It is a special lifecycle npm script that husky relies on https://typicode.github.io/husky/#/?id=install

Comment (nemanjaglumac) [reply package.json:208]: A note for the future (when we hopefully upgrade `yarn`):

> Yarn 2 doesn't support prepare lifecycle script, so husky needs to be installed differently (this doesn't apply to Yarn 1 though). See [Yarn 2 install](https://typicode.github.io/husky/#/?id=yarn-2).

**Pull request merged**